### PR TITLE
Edit history dialog

### PR DESCRIPTION
**Pull request by bwindels**
First part for https://github.com/vector-im/riot-web/issues/9490
Requires https://github.com/matrix-org/matrix-js-sdk/pull/967

edit-history2

### CODE REVIEW

File: src/components/views/dialogs/MessageEditHistoryDialog.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import {wantsDateSeparator} from '../../../DateUtils';
+import SettingsStore from '../../../settings/SettingsStore';
+
+export default class MessageEditHistoryDialog extends React.Component {
+    static propTypes = {
+        mxEvent: PropTypes.object.isRequired,
+    };
+
+    componentWillMount() {
+        this.loadMoreEdits = this.loadMoreEdits.bind(this);
```
Comment (jryans) [inline src/components/views/dialogs/MessageEditHistoryDialog.js]: Can you use [property initializer](https://github.com/matrix-org/matrix-react-sdk/blob/develop/code_style.md#react) style instead?

```js
loadMoreEdits = async (backwards) => {

}
```

File: src/components/views/dialogs/MessageEditHistoryDialog.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        this.setState({
```
Comment (jryans) [inline src/components/views/dialogs/MessageEditHistoryDialog.js]: This seems like initial state that would be [more naturally](https://github.com/matrix-org/matrix-react-sdk/blob/develop/code_style.md#react) handled in the constuctor:

```js
constructor(props) {
    super(props);
    this.state = { ... };
}
```

File: src/components/views/dialogs/MessageEditHistoryDialog.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import MatrixClientPeg from "../../../MatrixClientPeg";
+import { _t } from '../../../languageHandler';
+import sdk from "../../../index";
+import {wantsDateSeparator} from '../../../DateUtils';
+import SettingsStore from '../../../settings/SettingsStore';
+
+export default class MessageEditHistoryDialog extends React.Component {
```
Comment (jryans) [inline src/components/views/dialogs/MessageEditHistoryDialog.js]: I think it only depends on props and state, so could we use `PureComponent` here?

File: src/components/views/dialogs/MessageEditHistoryDialog.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        if (backwards || (!this.state.nextBatch && !this.state.isLoading)) {
+            // bail out on backwards as we only paginate in one direction
+            return false;
+        }
+        const opts = {token: this.state.nextBatch};
+        const roomId = this.props.mxEvent.getRoomId();
+        const eventId = this.props.mxEvent.getId();
+        const result = await MatrixClientPeg.get().relations(
+            roomId, eventId, "m.replace", "m.room.message", opts);
+        //console.log(`loadMoreEdits: got ${result.}`)
```
Comment (jryans) [inline src/components/views/dialogs/MessageEditHistoryDialog.js]: Maybe remove?

File: src/components/views/elements/EditHistoryMessage.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
+/*
```
Comment (jryans) [inline src/components/views/elements/EditHistoryMessage.js]: Would the `messages` directory be a better fit? (The component directories are all a bit haphazard...)

File: src/components/views/elements/EditHistoryMessage.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+Copyright 2019 New Vector Ltd
```
Comment (jryans) [inline src/components/views/elements/EditHistoryMessage.js]: Probably the copyright can be only foundation for a new file?

File: src/components/views/elements/EditHistoryMessage.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+limitations under the License.
+*/
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as HtmlUtils from '../../../HtmlUtils';
+import {formatTime} from '../../../DateUtils';
+import {MatrixEvent} from 'matrix-js-sdk';
+import {pillifyLinks} from '../../../utils/pillify';
+
+export default class EditHistoryMessage extends React.Component {
```
Comment (jryans) [inline src/components/views/elements/EditHistoryMessage.js]: Can it be a `PureComponent`?

File: src/components/views/messages/TextualBody.js [hunk 8]
```diff
@@ -468,6 +374,7 @@ module.exports = React.createClass({
         return (
             <div
                 key="editedMarker" className="mx_EventTile_edited"
+                onClick={this._openHistoryDialog}
                 onMouseEnter={this._onMouseEnterEditedMarker}
                 onMouseLeave={this._onMouseLeaveEditedMarker}
             >{editedTooltip}<span>{`(${_t("edited")})`}</span></div>
```
Comment (jryans) [inline src/components/views/messages/TextualBody.js]: Zeplin suggests this should say something like "Edited at 12:00. Click to view edits." Unsure if we have the edit time here, but the click instruction should likely be added.

File: src/utils/pillify.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
+import ReactDOM from 'react-dom';
```
Comment (jryans) [inline src/utils/pillify.js]: Add license header.

File: src/utils/pillify.js [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+import MatrixClientPeg from '../MatrixClientPeg';
+import SettingsStore from "../settings/SettingsStore";
+import PushProcessor from 'matrix-js-sdk/lib/pushprocessor';
+import sdk from '../index';
+
+
```
Comment (jryans) [inline src/utils/pillify.js]: Nit: Remove duplicate blank line

File: src/components/views/messages/TextualBody.js (hunk 8 above)
Comment (bwindels) [reply src/components/views/messages/TextualBody.js]: It already says "Edited at 12:00" actually, will add the "Click to view edits." part though :+1:

**Pull request merged**